libp2p/switch.nim

```diff
@@ -465,6 +465,12 @@ proc stop*(s: Switch) {.async.} =
   trace "switch stopped"
 
 proc muxerHandler(s: Switch, muxer: Muxer) {.async, gcsafe.} =
+  # store incoming connection
```

Comment:
@zah this isn't correct, identify *has* to happen before adding the connection. Identify performs peer verification and if that fails, the connection should be terminated and the upgrade stopped.

Reply:
I figured, but we can try to undo the registration in that case perhaps? The alternative that I considered is creating a future for the muxer record, so any dials will wait for it to be properly registered.

Reply:
technically, we verify the peer in noise already, as part of the handshake - ie that's where peer identity is first encountered / set up

Reply:
ie in the peerid branch from before, that's one of the things that were firmed up, that there are fewer peerinfo objects flying around - that said, noise creates one for the peer with the right public key which gives an identity which is enough to work with on the muxer side

Reply:
We should not create a new peerinfo object either way, we should reuse/update the one that is already setup. 

We might be able to get away with running identify after muxing, but that is definitely not the expected flow.

I've also ran this changes in medalla and it doesn't seem to improve by much, which tells me that there is something else going on.

Reply:
> We should not create a new peerinfo object either way

this would require some pretty major refactoring - I started it in the peerid branch but it takes a bit of work as it would require a centralized repository of peerinfo objects

Reply:
> undo the registration

the registration should ideally be followed by announcing the connection to the world, so that nbc for example can do its status dance